### PR TITLE
Word Bazaar Error Handling

### LINKED ISSUE #955: voting votes loading indicator

### LINKED ISSUE #956: voting info loading indicator

### LINKED ISSUE #957: overall error handling and communication

### LINKED ISSUE #966: WordBazzar - vote/apply/revoke/withdraw loading/steps design

### PR DESCRIPTION
**Pull request by AtanasKrondev**
Closes #955 
Closes #956 
Closes #957 
Closes #966

### CODE REVIEW

**Review by thepiwo — COMMENTED** [review]
looks good in general, whats the reason its still draft?

did you talk to @onvisions regarding design?

maybe we can somehow do this globally without try catch everywhere

File: src/components/AlertModal.vue [hunk 1]
```diff
@@ -27,8 +33,10 @@ export default {
   props: {
     title: { type: String, required: true },
     body: { type: String, required: true },
+    secondRow: { type: String, default: '' },
```
Comment (davidyuk) [inline src/components/AlertModal.vue:36]: `secondBody`. Maybe we can allow the alert modal to accept the body as an array of strings instead?

File: src/components/AlertModal.vue [hunk 2]
```diff
@@ -26,9 +34,10 @@ import FailureIcon from '../assets/iconError.svg';
 export default {
   props: {
     title: { type: String, required: true },
-    body: { type: String, required: true },
+    body: { type: Array, required: true },
```
Comment (davidyuk) [inline src/components/AlertModal.vue:37]:
```suggestion
    body: { type: [Syring, Array], required: true },
```

File: src/components/AlertModal.vue [hunk 3]
```diff
@@ -5,9 +5,17 @@
...
       <div class="success-modal">
-        <img :src="failure ? FailureIcon : SuccessIcon">
+        <img
+          v-if="!hideIcon"
+          :src="failure ? FailureIcon : SuccessIcon"
+        >
         <h1>{{ title }}</h1>
-        <p>{{ body }}</p>
+        <p
+          v-for="(row, index) in body"
```
Comment (davidyuk) [inline src/components/AlertModal.vue:14]:
```suggestion
          v-for="(row, index) in body instanceof Array ? body : [body]"
```

File: src/store/plugins/modals.js [hunk 4]
```diff
@@ -14,7 +14,8 @@ export default (store) => {
     state: { opened: [] },
     mutations: {
       open(state, modal) {
-        state.opened.push(modal);
+        state.opened
+          .push({ ...modal, props: { ...modal.props, body: [].concat(modal.props.body) } });
```
Comment (davidyuk) [inline src/store/plugins/modals.js:18]: Not all modals have a `body`, and some modals can expect a different type of `body`. So, this part shouldn't be aware of props that a specific modal accepts, let's revert this.

**Pull request merged**